spring-kafka/src/main/java/org/springframework/kafka/retrytopic/EndpointCustomizerFactory.java

```diff
@@ -71,7 +74,12 @@ protected EndpointCustomizer addSuffixesAndMethod(DestinationTopic.Properties pr
 			Collection<EndpointCustomizer.TopicNamesHolder> topics = customizeAndRegisterTopics(namesProvider, endpoint);
 			endpoint.setId(namesProvider.getEndpointId(endpoint));
 			endpoint.setGroupId(namesProvider.getGroupId(endpoint));
-			endpoint.setTopics(topics.stream().map(EndpointCustomizer.TopicNamesHolder::getCustomizedTopic).toArray(String[]::new));
+			if (endpoint.getTopics().isEmpty() && endpoint.getTopicPartitionsToAssign() != null) {
+				endpoint.setTopicPartitions(getTopicPartitions(properties, namesProvider, endpoint.getTopicPartitionsToAssign()));
```

Comment:
This breaks the initial offset on the main listener because it overwrites the existing ones with a `null` initial offset.

We shouldn't do this for the `mainEndpoint` in `processAndRegisterEndpoint`.

```java
@SpringBootApplication
public class Kgh2220Application {

	public static void main(String[] args) {
		SpringApplication.run(Kgh2220Application.class, args);
	}

	@Bean
	public NewTopic topic() {
		return TopicBuilder.name("kgh2220").partitions(1).replicas(1).build();
	}

	@KafkaListener(id = "kgh2220xxx", topicPartitions = @TopicPartition(topic = "kgh2220",
			partitionOffsets = @PartitionOffset(partition = "0", initialOffset = "0")))
	@RetryableTopic
	void listen(String in, @Header(KafkaHeaders.RECEIVED_TOPIC) String topic,
			@Header(KafkaHeaders.OFFSET) long offset) {
		System.out.println(in + "@" + offset);
		System.out.println(topic);
		System.out.println(KafkaUtils.getConsumerGroupId());
		throw new RuntimeException();
	}

	@Bean
	ApplicationRunner runner(KafkaTemplate<String, String> template) {
		return args -> {
			template.send("kgh2220", "foo");
		};
	}

}
```
```
foo@15
kgh2220
kgh2220xxx
```

Reply:
Good catch, makes sense. I'll fix it shortly, thanks.